src/main/scala/com/gossiperl/client/actors/ActorRegistry.scala

```diff
@@ -0,0 +1,60 @@
+package com.gossiperl.client.actors
+
+import akka.actor.{ActorRef, Actor}
+import com.typesafe.scalalogging.slf4j.LazyLogging
+
+import scala.concurrent.{Future, Promise}
+
+class ActorNotRegisteredException(message: String) extends RuntimeException(message)
+
+object ActorRegistry extends LazyLogging {
```

Comment:
might be interesting to have a look at the Akka built-in ActorRegistry. Apparently all actors are automatically registered to it when they are started, removed or stopped. Seems to do something similar as what you are trying to achieve here. http://doc.akka.io/docs/akka/1.3.1/scala/actor-registry.html